fpx-lib/src/api/models.rs

```diff
@@ -140,3 +140,32 @@ impl From<SpanAdded> for ServerMessage {
         ServerMessageDetails::SpanAdded(Box::new(val)).into()
     }
 }
+
+#[derive(Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct TypeScriptCompatSpan {
+    pub span_id: Option<String>,
+    pub trace_id: Option<String>,
+    pub created_at: time::OffsetDateTime,
+    pub updated_at: time::OffsetDateTime,
+    pub parsed_payload: Span,
+}
+
+impl Into<TypeScriptCompatSpan> for crate::data::models::Span {
+    fn into(self) -> TypeScriptCompatSpan {
+        TypeScriptCompatSpan {
+            span_id: Some(self.span_id.clone()),
+            trace_id: Some(self.trace_id.clone()),
+            created_at: self.start_time.into(),
```

Comment:
Think for both the created_at and updated_at we should just use the same timestamp. Since there really isn't a situation in which we can update a span. I think having both set to the end_time makes the same most sense

Reply:
Makes sense, will update!